### PR TITLE
Transform data

### PR DESCRIPTION
**Pull request by Annie-EXE**
Hello,

I added a 'transform' function which creates a datetime timestamp column when given a date string column and a time string column. I also added some tests for the transform file.

### CODE REVIEW

File: transform.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
+"""Pipeline Script: Transforming pipeline data"""
+
+from datetime import datetime
+import pandas as pd
+from pandas import DataFrame
+
+
+def load_data(csv_path: str) -> DataFrame:
```
Comment (NgawangT) [inline transform.py:8]: [PEP 257](https://peps.python.org/pep-0257/) describes good docstring conventions https://peps.python.org/pep-0008/#documentation-strings . Tiny note that the multiline docstring should be on the same line for opening docstring and then on a line by itself for e.g :
"""Return a foobang

Optional plotz says to frobnicate the bizbaz first.
"""

File: transform.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        with open(csv_path, 'r') as file:
+            data = pd.read_csv(csv_path)
+            return data
+
+    except (FileNotFoundError) as e:
+        print(f"Error loading .csv data: {e}")
+        return None
+    
+
+def hhmm_to_timestamp(hhmmss_time_string: str):
```
Comment (NgawangT) [inline transform.py:23]: check docstring above.

File: transform.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        dt = datetime.now().replace(hour=hh, minute=mm, second=ss, microsecond=0)
+        timestamp = dt.timestamp()
+        return timestamp
+
+    except ValueError:
+        print("Valid 'HHMMSS' time string must be provided")
+        return None
+
+
+def create_timestamp_from_date_and_time(df: DataFrame, new_column_name: str,
```
Comment (NgawangT) [inline transform.py:41]: check comment about docstring above.

**Review by HarryEP — APPROVED** [review]

File: transform.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    try:
+        df[time_column_name] = pd.to_datetime(df[time_column_name], format='%H%M%S')
+    
+    except ValueError as e:
+        print(f"Error: invalid values in time column")
+        return None
+    
+    df[new_column_name] = df[date_column_name] + pd.to_timedelta(df[time_column_name].dt.strftime('%H:%M:%S'))
+    return df
+
```
Comment (NgawangT) [inline transform.py:65]: It's great that you have included, Error handling for all your code.

Comment (NgawangT) [reply transform.py:65]: Clean code.

File: test_transform.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    is creating datetime objects in the new column,
+    with the required format (YYYY-MM-DD HH:MM:SS)
+    """
+    df = generate_test_dataframe_with_date_and_time_columns()
+    df = create_timestamp_from_date_and_time(df, "datetime",
+                                             "date_column", "time_column")
+    
+    datetime_value = df.loc[0, "datetime"]
+    assert datetime_value == pd.Timestamp("2023-09-05 12:30:45")
+    assert isinstance(datetime_value, pd.Timestamp)
```
Comment (NgawangT) [inline test_transform.py:82]: Good main tests and Doc strings.

**Pull request merged**